README.md

```diff
@@ -1,86 +1,118 @@
-```
-This builds dependencies in the subdirectory named depend and then builds MIGraphX using these dependencies.
+You can manually download and installing the above dependencies one by one. For convience, we provide a shell 
+script [install_prereqs.sh](./tools/install_prereqs.sh) that can automatically install all the above dependencies with
+the command 
+
+```./tools/install_prereqs.sh```
 
-## Building MIGraphX from source
+(Note: By default, all dependencies are installed at the default location `/usr/local` 
+and are accessible by all users. For the default location, `sudo` is required to run the script.
+You can also specify a location at which the dependencies are installed with `./tools/install_prereqs.sh $your_loc`.)
 
-## Configuring with cmake
+##### Building MIGraphX source and install libs
 
-First create a build directory:
+With the above dependencies installed, we can build source as:
+
+1) Go to the project folder and create a build directory:
 
 
 ```
-mkdir build; 
-cd build;
+mkdir build
+cd build
 ```
 
-Next configure cmake. The hcc compiler is required to build the MIOpen backend:
-
+2) Configure the cmake. If the dependencies are installed at the default location `/usr/local`, the command is:
 
 ```
 CXX=/opt/rocm/llvm/bin/clang++ cmake ..
 ```
+Otherwise, you need to set `-DCMAKE_PREFIX_PATH=$your_loc` to configure the cmake. 
 
-If the dependencies from `install_deps.cmake` was installed to another directory, the `CMAKE_PREFIX_PATH` needs to be set to what `--prefix` was set to from `install_deps.cmake`:
-
+3) Build MIGraphX source code
 
 ```
-CXX=/opt/rocm/llvm/bin/clang++ cmake -DCMAKE_PREFIX_PATH=/some/dir ..
+make -j$(nproc)
 ```
 
+Correctness can be verified as:
 
-#### Changing the cmake configuration
-
-The configuration can be changed after running cmake by using `ccmake`:
-
-` ccmake .. ` **OR** `cmake-gui`: ` cmake-gui ..`
-
-## Building the library
-
-The library can be built, from the `build` directory using the 'Release' configuration:
+```
+make -j$(nproc) check
+```
 
-` cmake --build . --config Release ` **OR** ` make `
+MIGraphX libs can be installed as:
 
-And can be installed by using the 'install' target:
+```
+make install
+```
 
-` cmake --build . --config Release --target install ` **OR** ` make install `
+#### Use the ROCm build tool [rbuild](https://github.com/RadeonOpenCompute/rbuild).
 
-This will install the library to the `CMAKE_INSTALL_PREFIX` path that was set.
+In this approach, we need to install the [rbuild](https://github.com/RadeonOpenCompute/rbuild) first, then use it to
+build MIGraphX. rbuild can be installed as (sudo may be needed.):
+```
+pip3 install https://github.com/RadeonOpenCompute/rbuild/archive/master.tar.gz
+```
+and pip3 can be installed as `sudo apt update && sudo apt install -y python3-pip`.
+We also need to install [rocm-cmake](https://github.com/RadeonOpenCompute/rocm-cmake) as `sudo apt install -y rocm-cmake`.
```

Comment:
More than just `rocm-cmake` is needed. They need to install both `miopen-hip` and `rocblas` as well.

Reply:
There should be a seperate step that describes which rocm packages need to be installed. Also, this should clarify that these are the packages for ubuntu.

Reply:
The reason I wrote it in this way is: rocm-cmake is needed before we can run the rbuild. As for the miopen and rocblas, they are installed in the script. I will not mention rocm-cmake here and move it to the script file, so it will be installed in the same way as the miopen and rocblas.